lib/assignDisabledRanges.mjs

```diff
@@ -61,101 +73,162 @@ export default function assignDisabledRanges(root, result) {
 
 	const configurationComment = result.stylelint.config?.configurationComment;
 
-	root.walkComments((comment) => {
-		if (inlineEnd) {
-			// Ignore comments already processed by grouping with a previous one.
-			if (inlineEnd === comment) inlineEnd = null;
+	root.walk((node) => {
+		if (isComment(node)) {
+			if (inlineEnd) {
+				// Ignore comments already processed by grouping with a previous one.
+				if (inlineEnd === node) inlineEnd = null;
 
-			return;
-		}
+				return;
+			}
 
-		const nextComment = comment.next();
+			const nextComment = node.next();
+
+			// If any of these conditions are not met, do not merge comments.
+			if (
+				!(
+					!isStandardSyntaxComment(node) &&
+					isConfigurationComment(node.text, configurationComment) &&
+					nextComment &&
+					isComment(nextComment) &&
+					(node.text.includes('--') || nextComment.text.startsWith('--'))
```

Comment:
once it's ready to be implemented

Could you put `(node.text.includes('--') || nextComment.text.startsWith('--')` in a `const`?
This would serve as documentation.
i.e. it's an optional disable description
e.g. `hasDisableDescription`

medium
fix

> There must be a space on both sides of the hyphens.

Hence, it should be `' -- '`, not `'--'`.

Reply:
This is unrelated to this change :)
Only the indent changed and variable names changed, I didn't make any logic changes to this part.

Do you mind filing a separate issue for this?

Reply:
> Do you mind filing a separate issue for this?

If it doesn't get the `ready to implement` label, I will.
Else you could just add another changeset for the `' -- '` fix.

Reply:
I prefer not to make that change here, as it is completely unrelated.

Reply:
The regex would be too complicated because javascript doesn't support the `\Z` metacharacter.